scipy/special/_binom.h

```diff
@@ -0,0 +1,88 @@
+// Binomial coefficient
+#pragma once
+
+#include <cmath>
+#include <limits>
+
+#include "cephes.hh"
+
+
+using std::numeric_limits;
```

Comment:
I don't think we should import things like this in header files. Best to keep it fully qualified.

If you really want to do `using std::numeric_limits;`, you can actually do that within either the functions or namespaces it is used in. But I'd stick with not doing it.

Reply:
Ah, good catch. Agreed on not wanting to have unexpected side effects when including a header. There are some imports in `_lambertw.h` that can be removed in another PR.

https://github.com/scipy/scipy/blob/4d790c14456f72ef7a56aafa0dcc728e5bba22fe/scipy/special/_lambertw.h#L32-L34